front-end/src/screens/Settings/address.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Icon } from 'semantic-ui-react';
+import { web3Accounts, web3FromSource, web3Enable } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+import { stringToHex } from '@polkadot/util';
+import Identicon from '@polkadot/react-identicon';
+
+import getExtensionUrl from '../../util/getExtensionUrl';
+import { NotificationContext } from '../../context/NotificationContext';
+import { useAddressesQuery, useAddressLinkConfirmMutation, useAddressLinkStartMutation, useAddressUnlinkMutation } from '../../generated/graphql';
+import Button from '../../ui-components/Button';
+import { NotificationStatus } from '../../types';
+
+const APP = 'polkassembly';
+
+const Address = (): JSX.Element => {
+	const [linked, setLinked] = useState<{[key: string]: boolean}>({});
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [addressLinkStartMutation] = useAddressLinkStartMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const [addressLinkConfirmMutation] = useAddressLinkConfirmMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const [addressUnlinkMutation] = useAddressUnlinkMutation({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const { data } = useAddressesQuery({ context: { uri : process.env.REACT_APP_AUTH_SERVER_GRAPHQL_URL } });
+	const { queueNotification } = useContext(NotificationContext);
+
+	useEffect(() => {
+		async function connect() {
+			await web3Enable(APP);
+
+			const allAccounts = await web3Accounts();
+
+			setAccounts(allAccounts);
+		}
+
+		connect();
+	}, []);
+
+	useEffect(() => {
+		data && data.addresses && data.addresses.forEach((address) => {
```

Comment:
use `{address}` as parameter so that you don't have this weird `address.address`

Reply:
fixed. but typescript still complaining it could be undefined. So added extra checks.